packages/server/src/unstable-core-do-not-import/stream/sse.test.ts

```diff
@@ -139,3 +141,162 @@ test('e2e, server-sent events (SSE)', async () => {
   await server.close();
   expect(values).toEqual(range(1, ITERATIONS * 2 + 1));
 });
+
+test('SSE on serverless - emit and disconnect early', async () => {
+  async function* data(lastEventId?: Maybe<number>) {
+    let i = lastEventId ?? 0;
+
+    function* yieldEvent() {
+      i++;
+      yield {
+        id: i,
+        data: i,
+      } satisfies SSEvent;
+    }
+    while (true) {
+      // yield 2 events at a time to test if the client will get both without reconnecting in between
+      yield* yieldEvent();
+      yield* yieldEvent();
```

Comment:
Just out of curiosity, why are you forwarding the yield here? Since `yieldEvent` only yields once, this looks equivalent to a regular function call, isn't it?
```suggestion
    function yieldEvent() {
      i++;
      return {
        id: i,
        data: i,
      } satisfies SSEvent;
    }
    while (true) {
      // yield 2 events at a time to test if the client will get both without reconnecting in between
      yield yieldEvent();
      yield yieldEvent();
```

(TBH I've never actually had a use-case for forwarding an iterator with `yield*` so I might be missing a nuance here)

Reply:
Just felt cool to use that feature haha